brave/brave-core
# Menu items for opening Tor windows

fix #1387

- Main hamburger menu:
  New private window with Tor

- Context menu for right-click on links:
  Open link in private window with Tor

This pushes IsTorProfile() into BrowserContext where it now resides
alongside IsOffTheRecord(), since we need it in order to determine
whether the 'Open link in private window with Tor' item should be
greyed out or not.  (Is this a little silly?  Maybe, but it's what the
browser already does with 'Open link in incognito window' item.)

Test plan:

1. 1. Open a fresh browser instance.
   2. Confirm that hamburger -> 'New private window with Tor' (NPWWT)
      opens a private window with Tor.
   3. Confirm that in the _normal_ window's hamburger, NPWWT opens a
      _new_ private window with Tor.
   4. Confirm that in the _Tor_ window's hamburger, NPWWT opens a _new_
      private window with Tor.

2. 1. Open a fresh browser instance.
   2. NPWWT
   3. In the Tor window, open <https://check.torproject.org/>.
   4. Confirm that you are using Tor.
   5. Right-click any link on the page.
   6. Confirm that items to open in a private window, with or without
      Tor, are greyed out.

3. 1. Open a fresh browser instance.
   2. In a _normal_ window, open <https://www.example.com/>.
   3. Right-click the 'More information...' link.
   4. Select 'Open link in private window with Tor' (OLIPWWT).
   5. Confirm that the link opens in a Tor window.
   6. In the normal window, do it again.
   7. Confirm that the link opens in a new tab in the existing Tor
      window.

TODO: Rename IsTorProfile to make more sense for BrowserContext.
TODO: Reuse rather than copy OpenEmptyWindow in browser_commands.cc.

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on

## Test Plan:


## Reviewer Checklist:
